clients/src/test/java/org/apache/kafka/clients/producer/internals/SenderTest.java

```diff
@@ -1196,6 +1197,47 @@ public void testResetOfProducerStateShouldAllowQueuedBatchesToDrain() throws Exc
         assertEquals(0, transactionManager.sequenceNumber(tp1).longValue());
     }
 
+    @Test
+    public void testCloseWithProducerIdReset() throws Exception {
```

Comment:
Could we have a separate test case which covers the `forceClose()` path. We can assert that we do not wait for the producerId and all pending sends are aborted.

Reply:
@hachikuji Thanks for the review. Addressed the review comment.